crates/forge/bin/cmd/clone.rs

```diff
@@ -0,0 +1,674 @@
+        ([$($key:expr),+], $value:expr) => {
+            {
+                if let Some(value) = $value {
+                    let mut current = &mut doc[Config::PROFILE_SECTION][profile];
+                    $(
+                        if let Some(nested_doc) = current.get_mut(&$key) {
+                            current = nested_doc;
+                        } else {
+                            return Err(eyre::eyre!("cannot find the key: {}", $key));
+                        }
+                    )+
+                    *current = toml_edit::value(value);
+                }
+            }
+        };
+    }
+
+    // update the chain id
+    doc[Config::PROFILE_SECTION][profile]["chain_id"] = toml_edit::value(chain.id() as i64);
+
+    // disable auto detect solc and set the solc version
+    doc[Config::PROFILE_SECTION][profile]["auto_detect_solc"] = toml_edit::value(false);
+    let version = meta.compiler_version()?;
+    doc[Config::PROFILE_SECTION][profile]["solc_version"] =
+        toml_edit::value(format!("{}.{}.{}", version.major, version.minor, version.patch));
+
+    // get optimizer settings
+    // XXX (ZZ): we ignore `model_checker`, `debug`, and `output_selection` for now,
+    // it seems they do not have impacts on the actual compilation
+    let Settings { optimizer, libraries, evm_version, via_ir, stop_after, metadata, .. } =
+        meta.settings()?;
+    eyre::ensure!(stop_after.is_none(), "stop_after should be None");
+
+    update_if_needed!(["evm_version"], evm_version.map(|v| v.to_string()));
+    update_if_needed!(["via_ir"], via_ir);
+
+    // update metadata if needed
+    if let Some(metadata) = metadata {
+        update_if_needed!(["cbor_metadata"], metadata.cbor_metadata);
+        update_if_needed!(["use_literal_content"], metadata.use_literal_content);
+        update_if_needed!(["bytecode_hash"], metadata.bytecode_hash.map(|v| v.to_string()));
+    }
+
+    // update optimizer settings if needed
+    update_if_needed!(["optimizer"], optimizer.enabled);
+    update_if_needed!(["optimizer_runs"], optimizer.runs.map(|v| v as i64));
+    // update optimizer details if needed
+    if let Some(detail) = optimizer.details {
+        doc[Config::PROFILE_SECTION][profile]["optimizer_details"] = toml_edit::table();
+
+        update_if_needed!(["optimizer_details", "peephole"], detail.peephole);
+        update_if_needed!(["optimizer_details", "inliner"], detail.inliner);
+        update_if_needed!(["optimizer_details", "jumpdestRemover"], detail.jumpdest_remover);
+        update_if_needed!(["optimizer_details", "orderLiterals"], detail.order_literals);
+        update_if_needed!(["optimizer_details", "deduplicate"], detail.deduplicate);
+        update_if_needed!(["optimizer_details", "cse"], detail.cse);
+        update_if_needed!(["optimizer_details", "constantOptimizer"], detail.constant_optimizer);
+        // XXX (ZZ): simpleCounterForLoopUncheckedIncrement seems not supported by fourndry
```

Comment:
could you open an issue for this with a link to the docs, we likely just miss bindings for this setting

Reply:
Sure thing! I've opened an [issue](https://github.com/foundry-rs/compilers/issues/99). Since this seems directly related to the `compilers` repo, I do not report it in the `foundry` repo.

Reply:
Wait for the release of `foundry-compilers` 0.3.16 to include this update.

Reply:
this should be available now